packages/nimble-components/src/wafer-map/tests/hover-handler.spec.ts

```diff
@@ -0,0 +1,78 @@
+import { zoomIdentity } from 'd3-zoom';
+import { tableFromArrays } from 'apache-arrow';
+import { html } from '@microsoft/fast-element';
+import { HoverHandler } from '../modules/experimental/hover-handler';
+import { WaferMapOriginLocation } from '../types';
+import {
+    getDataManagerMockForHover,
+    getScaleQuantile,
+    getWaferMapMockHover
+} from './utilities';
+import type { WaferMap } from '..';
+import type { DataManager } from '../modules/data-manager';
+import { processUpdates } from '../../testing/async-helpers';
+import { Fixture, fixture } from '../../utilities/tests/fixture';
+
+async function setup(): Promise<Fixture<HTMLDivElement>> {
+    return fixture<HTMLDivElement>(html`<div></div>`);
+}
+
+describe('HoverHandler', () => {
+    let element: HTMLDivElement;
+    let connect: () => Promise<void>;
+    let disconnect: () => Promise<void>;
+    let hoverHandler: HoverHandler;
+    let waferMock: WaferMap;
+
+    beforeEach(async () => {
+        ({ element, connect, disconnect } = await setup());
+        await connect();
+        waferMock = getWaferMapMockHover(
+            tableFromArrays({
+                colIndex: Int32Array.from([1]),
+                rowIndex: Int32Array.from([1]),
+                value: Float64Array.from([1])
+            }),
+            zoomIdentity,
+            WaferMapOriginLocation.bottomLeft,
```

Comment:
We're missing some tests for particular code paths in `HoverHandler` because we're using one state like this in the test, when the implementation also cares about `bottomRight`. Additionally, we're not using data values (that I can see) that would be leveraging the `Math.ceil` or `Math.floor` functionality. It seems like we should be creating a `parameterizeSpec` that allows us to test multiple variants of state with expected results.

Reply:
I made some parametric tests, but it seems taht the scale quantile does not need floor or ceil. I will leave them there so we can switch it later with the data manager